src/charm.py

```diff
@@ -43,6 +45,66 @@ def __init__(self, *args: typing.Any) -> None:
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.reconciliation)
         self.framework.observe(self.on.spring_boot_app_pebble_ready, self.reconciliation)
+        self.database: DatabaseRequires = self._setup_database_requirer("mysql", "spring-boot")
+
+    def _setup_database_requirer(self, relation_name: str, database_name: str) -> DatabaseRequires:
+        """Set up a DatabaseRequires instance.
+
+        The DatabaseRequires instance is an interface between the charm and various data providers.
+        It handles those relations and emit events to help us abstract these integrations.
+
+        Args:
+            relation_name: Name of the data relation
+            database_name: Name of the database (can be overwritten by the provider)
+
+        Returns:
+            DatabaseRequires object produced by the data_platform_libs.v0.data_interfaces library
+        """
+        database_requirer = DatabaseRequires(
+            self,
+            relation_name=relation_name,
+            database_name=database_name,
+        )
+        self.framework.observe(database_requirer.on.database_created, self.reconciliation)
+        self.framework.observe(self.on[relation_name].relation_broken, self.reconciliation)
+        return database_requirer
+
+    def _datasource(self) -> dict[str, str]:
```

Comment:
would this be better served with `@property` decorator? or the name of the function to be a verb, such as `get_datasource_metadata`?
I think calling `self._datasource()` looks a little awkward for me :)

Reply:
It's a private function not meant to be usable outside of the class. That's why we prefix it with an underscore. The same idea applies in my opinion for the `@property` decorator that would signify that this is usable outside of the class (which is not the case here).